packages/lest/src/lua/utils/is-lua-symbol.lua

```diff
@@ -0,0 +1,34 @@
+local KEYWORDS = {
+	["and"] = true,
+	["break"] = true,
+	["do"] = true,
+	["else"] = true,
+	["elseif"] = true,
+	["end"] = true,
+	["false"] = true,
+	["for"] = true,
+	["function"] = true,
+	["if"] = true,
+	["in"] = true,
+	["local"] = true,
+	["nil"] = true,
+	["not"] = true,
+	["or"] = true,
+	["repeat"] = true,
+	["return"] = true,
+	["then"] = true,
+	["true"] = true,
+	["until"] = true,
+	["while"] = true,
+}
+
+--- Returns true if the value is a string containing a valid Lua symbol
+---@param value any
+---@return boolean
+local function isLuaSymbol(value)
+	return type(value) == "string"
+		and not KEYWORDS[value]
+		and not not string.match(value, "^[_%a][_%a%d]*$")
```

Comment:
Just to confirm, this is a boolean coercion right? Looks sort of strange

Reply:
Yep, to avoid returning a non-boolean here